core/templates/services/translate.service.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Translate service for i18n translations
+ */
+
+import { Injectable, EventEmitter } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { UtilsService } from './utils.service';
+
+export interface MissingTranslationHandlerParams {
+   // the key that's missing in translation files
+  key: string;
+
+  // an instance of the service that was unable to translate the key.
+  translateService: TranslateService;
+
+   // Interpolation params that were passed along for translating the given key.
+  interpolateParams?: Object;
+}
+
+export interface LangChangeEvent {
+  lang: string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TranslateService {
+  currentLang = 'en';
+  defaultLang = 'en';
+  prefix = '/assets/i18n/';
+  suffix = '.json';
+  translations: Array<Object> = [];
+  templateMatcher: RegExp = /\<\[\s?([^{}\s]*)\s?\]\>/g;
+
+  private _onLangChange = new EventEmitter<LangChangeEvent>();
+  get onLangChange(): EventEmitter<LangChangeEvent> {
+    return this._onLangChange;
+  }
+
+  constructor(private http: HttpClient, private utils: UtilsService) {}
+
+  // This function fecthes translation.
```

Comment:
"fetches" is misspelled.

Also, write proper jsdoc. Explain format of return value, etc.